### ISSUE TITLE
PeepCode 0.3.x lags/doesn't show files when searching

### ISSUE THREAD

**Issue by rufo** [topfunky/PeepOpen-Issues#216]
PeepOpen has been unusable for me since 0.3.0; I held out for the 0.3.1 update, but it doesn't seem to help.

Launching PeepOpen from the Finder is fine, as well as changing any of the settings. However, when I try to activate it from within a project in TextMate, it claims to scan the files, the progress bar disappears, but PeepOpen never actually shows anything in the list, and the "Loading…" text at the bottom remains. Sometimes the window gets focus and I can type stuff in (albeit with a huge lag, incredibly slow searching and nothing appearing in the list), but sometimes it seems not to. Generally speaking PeepCode seems out to lunch at that point and the best thing to do is force quit. The only clue I can figure out are a dozen to two dozen messages like these every time I activate PeepCode to do a search:

```
3/25/11 4:05:59 PM  PeepOpen[79102] /Applications/PeepOpen.app/Contents/Resources/rb_main.rb:44:in `<main>': no implicit conversion from nil to integer (TypeError)
```

I've dabbled in MacRuby some in the past, and for a while I'd compiled the source by hand due to some issues with the newer BridgeSupport/0.8 code. I tried to open up the 0.3.1 source, but Xcode 4.0 seems really unhappy right now and it just crashes with an inconsistency error rather quickly when I open that project.

Sorry, I know this is random - any ideas? I have a feeling it might be something on my machine since I haven't seen anything else, but I certainly don't know for sure…

**Comment by topfunky** [comment]
Thanks for the detailed bug report. I'm working on fixing similar issues right now.

It sounds like you may have an older version of the TextMate plugin. Try re-installing the TextMate plugin from PeepOpen's Preferences window, then quit and restart TextMate.

Or, try this from the command line (with a path to a project):

```
open 'peepopen://Users/my-user/path/to/project?editor=TextMate'     
```

Does that show anything?

**Comment by rufo** [comment]
I've re-installed the TextMate plugin several times with no effect, and the URL scheme from Terminal causes PeepOpen to behave similarly I'm afraid.

Thanks for the thoughts; let me know if there's anything else you'd like me to try.

**Comment by topfunky** [comment]
I just published an update that fixes bugs related to this issue (0.3.2).

Run "Check for Updates" or get it at PeepCode:

https://peepcode.com/products/peepopen

**Comment by rufo** [comment]
Zero change, I'm afraid. Auto-updated, verified I was running 0.3.2, re-installed TextMate plugin, tried with several projects, and I get exactly the same behavior.

**Comment by topfunky** [comment]
Hmm...well at least this fixes other issues. I'll try to reproduce the error you mentioned and figure out a fix.

**Comment by joshuabates** [comment]
Same problem here running 0.3.2 with the vim & textmate plugins. I've got macruby running locally as well.

**Comment by handcoding** [comment]
@rufo, @joshuabates: By chance do you have the AckMate plugin installed? If so, perhaps you may be seeing issue #229?

**Comment by rufo** [comment]
I double-checked this (removing both AckMate and an older Ack in Project bundle I had installed) and it still didn't work.

My spidey sense says that issue isn't quite related - it happens even when PeepOpen is launched from the shell without TextMate even being open, and as files never actually show up in the list, there's nothing for me to select - but I certainly could be wrong about that.

**Comment by jrk** [comment]
Same issue with 0.3.2 in MacRuby, TextMate, and directly via the URL scheme from the shell, including after wiping MacRuby installs, preferences, caches, and testing on a blank account.

**Comment by kemayo** [comment]
This is happening to me as well with 0.3.2 and TextMate. Large-ish project (bit over 7k files). When I open PeepOpen for the first time in a little while (e.g. after 30 minutes away not having to open any new files) it gets _really_ slow; 3-5 seconds between each keystroke registering, so typing in enough to find anything I'm looking for takes 10-30 seconds of waiting.

If there's any sort of logs / debugging data I can provide, let me know...

**Comment by kemayo** [comment]
Incidentally, I crashed PeepOpen while trying to open it from the command line to see if that would help. When it reopened and rescanned from inside TextMate I found it was a _lot_ snappier. Some sort of accumulated memory leak, perhaps?

**Comment by jrk** [comment]
@kemayo that is a different issue. This bug results in literally no feedback from the UI, ever, under any circumstances—not just slow performance. It seems related to the console.log line reported in the original (`no implicit conversion from nil to integer`…).

**Comment by topfunky** [comment]
I'm testing out a version that's much faster on my demo project of 10,000 files. It's been stable since last Friday, so I hope to publish an update early tomorrow.

**Comment by topfunky** [comment]
Version 0.3.3 and greater include some speed optimizations. The first character searched for in a large project will still take a few seconds, but the next characters will be almost instant. I've found this to make it much more usable with my demo project of 10,000 files. Try it out.

**Comment by rufo** [comment]
Yep, the newer versions still exhibit the original problem on my computer; I get the same errors in Console and nothing ever shows up in the PeepOpen file list. I'd take another stab at compiling it myself or looking into MacRuby/BridgeSupport stuff (though other MacRuby apps seem fine), but I'm neck-deep in another project at the moment… another day, perhaps

**Comment by jrk** [comment]
Indeed, same errors. (And I'd gladly compile and attach a debugger myself, but the source distribution seems quite far from being buildable out of the box.)

**Comment by rufo** [comment]
So I've upgraded to Lion this past week, and whatever changes have been made to my system have eliminated the problem - since I'm no longer affected, I'm afraid I can't comment on this issue anymore. It was definitely still a problem before then, though :)

**Comment by jrk** [comment]
Ditto.

**Comment by ebertech** [comment]
I've got the problem on 10.6 still.  Peepopen 0.3.5
